clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -1544,6 +1604,30 @@ bool SemaHLSL::CheckBuiltinFunctionCall(unsigned BuiltinID, CallExpr *TheCall) {
       return true;
     break;
   }
+  case Builtin::BI__builtin_hlsl_select: {
```

Comment:
This implementation has quite a different shape to the other builtins in this function.  I'm trying to figure out if there's something special about `select` that makes this necessary, or if this validation can be expressed using the existing CheckXXX functions in here.

Reply:
I think it is slightly special. It is a template function, and the first argument must be either a bool or a vector of bools.  Both of those cases have slightly different checks; so I made a function for each. I don't believe any of the existing Check functions checked for the correct things; most of them require all args are the same type.